### PR TITLE
Donate button

### PR DESCRIPTION
**Pull request by chasulin**
- Replaces social icons with donate button in AppBar
- Pulls social-icon code out into SocialIcons.tsx to add back in at a later time

### CODE REVIEW

File: src/components/AppBar/DonateButton.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      <ButtonContent />
+    </DonateButtonWrapper>
+  );
+};
+
+export const DonateButtonWithFade = () => {
+  // uses https://www.npmjs.com/package/@react-hook/window-scroll :
+  const scrollY = useScrollPosition(5);
+  // 170 approximates the height of the donation banner:
+  const isPassedBanner = scrollY > 170;
```
Comment (pnavarrc) [inline src/components/AppBar/DonateButton.tsx:45]: The offset might be different in desktop / mobile

File: src/components/AppBar/DonateButton.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <DonateButtonWrapper>
+      <ButtonContent />
+    </DonateButtonWrapper>
+  );
+};
+
+export const DonateButtonWithFade = () => {
+  // uses https://www.npmjs.com/package/@react-hook/window-scroll :
+  const scrollY = useScrollPosition(5);
```
Comment (pnavarrc) [inline src/components/AppBar/DonateButton.tsx:43]: _nit_: Why did you set 5? The default value is probably ok

File: src/components/AppBar/SocialIcons.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
+/*
+Note (Chelsi):
+We're replacing social icons with a 'Donate' button for the time being.
+Keeping all social-icons-related code here for if/when we reinstate them.
+*/
```
Comment (pnavarrc) [inline src/components/AppBar/SocialIcons.tsx:5]: 👍

File: src/components/AppBar/AppBar.tsx [hunk 4]
```diff
@@ -84,6 +34,8 @@ const _AppBar = () => {
   const locationPath = useLocation();
   const { isEmbed } = useEmbed();
 
+  const isLocationPage = location.pathname.includes('us');
```
Comment (pnavarrc) [inline src/components/AppBar/AppBar.tsx]: If you want to detect the homepage, you can probably use `location.pathname.length === 1`

File: src/components/AppBar/AppBar.tsx [hunk 5]
```diff
@@ -92,6 +44,15 @@ const _AppBar = () => {
...
   }, [history]);
 
+  /*
+  Fade functionality is only applied to mobile homepage,
+  where the donate button doesn't appear on load--it appears
+  only once scrollY has passed 175px / banner is scrolled away
+  */
+  const MobileDonateButtonComponent = isLocationPage
+    ? DonateButtonWithoutFade
+    : DonateButtonWithFade;
```
Comment (pnavarrc) [inline src/components/AppBar/AppBar.tsx]: Consider moving the logic of which button to show to the component 

```tsx
<DonateButton fadeOnScroll={isHomepage} fadeScrollPosition={170} />
```

File: src/components/AppBar/DonateButton.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  // 170 approximates the height of the donation banner:
+  const isPassedBanner = scrollY > 170;
+
+  return (
+    <Fade in={isPassedBanner} timeout={150}>
+      <DonateButtonWrapper>
+        <ButtonContent />
+      </DonateButtonWrapper>
```
Comment (pnavarrc) [inline src/components/AppBar/DonateButton.tsx:51]: _nit_: You can use `DonateButtonWithFade` here to ensure that they remain consistent

**Review by pnavarrc — COMMENTED** [review]
Added some notes

File: src/components/AppBar/DonateButton.tsx (hunk 2 above)
Comment (chasulin) [reply src/components/AppBar/DonateButton.tsx:43]: the default is 30, and knocking it down to 5 made the hook fire less frequently. but i guess its not that big of a deal, so ill switch back to the default.

File: src/components/AppBar/DonateButton.tsx (hunk 1 above)
Comment (chasulin) [reply src/components/AppBar/DonateButton.tsx:45]: this version is only used on mobile

File: src/components/AppBar/DonateButton.tsx (hunk 6 above)
Comment (chasulin) [reply src/components/AppBar/DonateButton.tsx:51]: the wrapper applies to both versions (with and without fade), so i think that keeping it generic as `DonateButtonWrapper` makes sense

File: src/components/AppBar/AppBar.tsx (hunk 4 above)
Comment (chasulin) [reply src/components/AppBar/AppBar.tsx]: re: our convo on slack, this wouldnt work :(

File: src/components/Analytics/utils.ts [hunk 7]
```diff
@@ -88,3 +89,7 @@ export function trackShare(label: string) {
 export function trackVoteClick(label: string) {
   trackEvent(EventCategory.VOTE_2020, EventAction.CLICK_LINK, label);
 }
+
+export function trackClick(label: string) {
+  trackEvent(EventCategory.DONATE, EventAction.CLICK, label);
+}
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/components/Analytics/utils.ts:95]: ooo good call. will use `trackEvent`, and will keep all donate actions under the DONATE event category

**Review by pnavarrc — APPROVED** [review]
:shipit:

**Pull request merged**